src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,100 @@
+package core.basesyntax;
+
+import core.basesyntax.exception.RegistrationDataException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static User user;
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+
+    @BeforeAll
+    static void createValidUser() {
+        user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("validPassword");
+        user.setAge(33);
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        Assertions.assertThrows(RegistrationDataException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void register_nullUserLogin_notOk() {
+        user.setLogin(null);
+        Assertions.assertThrows(RegistrationDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_nullUserPassword_notOk() {
+        user.setPassword(null);
+        Assertions.assertThrows(RegistrationDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_nullUserAge_notOk() {
+        user.setAge(null);
+        Assertions.assertThrows(RegistrationDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_underAge_notOk() {
+        user.setAge(17);
+        Assertions.assertThrows(RegistrationDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        user.setLogin("short");
+        Assertions.assertThrows(RegistrationDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        user.setPassword("short");
+        Assertions.assertThrows(RegistrationDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_existingLogin_notOk() {
+        registrationService.register(user);
+        User anotherUser = new User();
+        anotherUser.setLogin("validLogin");
+        anotherUser.setPassword("anotherPassword");
+        anotherUser.setAge(22);
+        Assertions.assertThrows(RegistrationDataException.class, () -> {
+            registrationService.register(anotherUser);
+        });
```

Comment:
The test for registering a user with an existing login is not correct. The user is not added to a shared storage so the test will not actually test if duplicate logins are handled properly. Use a low-level method to add the user to the storage before trying to register another user with the same login.